--- ../models/Staticroute.php	2020-07-23 11:20:43.000000000 +0200
+++ ../models/Staticroute-modified.php	2020-07-27 22:53:36.544417777 +0200
@@ -37,6 +37,11 @@
      */
     public $name;

+     /**
+     * @var int
+     */
+    public $basePage;
+
     /**
      * @var string
      */
@@ -113,6 +118,13 @@
     protected static $nameIdMappingCache = [];

     /**
+     * this is a small per request cache which holds id => url basePage mappings
+     *
+     * @var array
+     */
+    protected static $basePageIdMappingCache = [];
+
+    /**
      * contains the static route which the current request matches (it he does), this is used in the view to get the current route
      *
      * @var Staticroute
@@ -242,11 +254,38 @@
     }

     /**
+     * @param boolean $prependBasePage
      * @return string
      */
-    public function getPattern()
+    public function getPattern($prependBasePage = false)
     {
-        return $this->pattern;
+        $pattern = $this->pattern;
+        if (!$prependBasePage) {
+            return $pattern;
+        }
+        $basePagePath = $this->getBasePagePath();
+        if (!$basePagePath) {
+            return $pattern;
+        }
+        // in case there is an empty pattern
+        if (!$pattern) {
+            return $basePagePath;
+        }
+
+        $delimiter = $pattern[0];
+        // if the delimiter is / insert basePagePath before first \/ (and escape / in basePagePath)
+        // otherwise insert basePagePath before first /
+        $firstSlashPosition = false;
+        if ($delimiter == '/') {
+            $firstSlashPosition = stripos($pattern, '\\/');
+            $basePagePath = str_replace('/', '\\/', $basePagePath);
+        } else {
+            $firstSlashPosition = stripos($pattern, '/');
+        }
+        if ($firstSlashPosition !== false) {
+            $pattern = substr_replace($pattern, $basePagePath, $firstSlashPosition, 0);
+        }
+        return $pattern;
     }

     /**
@@ -414,6 +453,49 @@
     }

     /**
+     * @return int
+     */
+    public function getBasePage()
+    {
+        return $this->basePage;
+    }
+
+    /**
+     * @param int $basePage
+     * @return void
+     */
+    public function setBasePage($basePage)
+    {
+        $this->basePage = $basePage;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getBasePagePath()
+    {
+        $basePagePath = null;
+        if ($this->getBasePage()) {
+            if (isset(self::$basePageIdMappingCache[$this->getBasePage()])) {
+                $basePagePath = self::$basePageIdMappingCache[$this->getBasePage()];
+            } else {
+                $document = Document::getById($this->getBasePage());
+                if ($document instanceof Document\Page) {
+                    $basePagePath = $document->getFullPath();
+                } elseif ($document instanceof Document\Link) {
+                    $basePagePath = $document->getHref();
+                } elseif ($document instanceof Document\Hardlink) {
+                    $basePagePath = $document->getFullPath();
+                }
+                self::$basePageIdMappingCache[$this->getBasePage()] = $basePagePath;
+            }
+        }
+        return $basePagePath;
+    }
+
+    /**
      * @param string $reverse
      *
      * @return $this
@@ -595,6 +677,12 @@
         // convert tmp urlencode escape char back to real escape char
         $url = str_replace($urlEncodeEscapeCharacters, '%', $url);

+        // preprend basePage
+        $basePagePath = $this->getBasePagePath();
+        if ($basePagePath) {
+            $url = $basePagePath.$url;
+        }
+
         $event = new GenericEvent($this, [
             'frontendPath' => $url,
             'params' => $urlParams,
@@ -617,7 +705,7 @@
      */
     public function match($path, $params = [])
     {
-        if (@preg_match($this->getPattern(), $path)) {
+        if (@preg_match($this->getPattern(true), $path)) {

             // check for site
             if ($this->getSiteId()) {
@@ -646,7 +734,7 @@

             $variables = explode(',', $this->getVariables());

-            preg_match_all($this->getPattern(), $path, $matches);
+            preg_match_all($this->getPattern(true), $path, $matches);

             if (is_array($matches) && count($matches) > 1) {
                 foreach ($matches as $index => $match) {
